src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -2813,6 +2814,30 @@ namespace winrt::TerminalApp::implementation
         }
     }
 
+    // Method Description:
+    // - Moves the tab to another index in the tabs row (if required).
+    // Arguments:
+    // - currentTabIndex: the current index of the tab to move
+    // - suggestedNewTabIndex: the new index of the tab, might get clamped to fit int the tabs row boundaries
+    // Return Value:
+    // - <none>
+    void TerminalPage::_TryMoveTab(const uint32_t currentTabIndex, const int32_t suggestedNewTabIndex)
+    {
+        uint32_t newTabIndex = std::clamp<int32_t>(suggestedNewTabIndex, 0, _tabs.Size() - 1);
```

Comment:
concern: technically this is a data-destructive cast, but we know that it's valid. Want to throw a `gsl::narrow_cast<uint32_t>` on it? That's the non-throwing variety :smile: